### ISSUE TITLE
Add support for the FID metric.

### ISSUE THREAD

**Issue by ohayonguy** [PyTorchLightning/pytorch-lightning#6959]
## 🚀 Feature
Add support for the Frechet Inception Distance as a metric, to evaluate generative models (https://arxiv.org/abs/1706.08500)

### Motivation
This metric is used in many recent GAN applications to evaluate the perceptual quality of the generated images.
It is therefore a very helpful and important metric to implement.

### Alternatives
The inception score could be also used as well, but FID is more acceptable today.

**Comment by awaelchli** [comment]
I have seen people implement it as a callback. Not sure how well this fits into the [torchmetrics](https://github.com/PyTorchLightning/metrics) framework, but maybe post an issue there too?

**Comment by ohayonguy** [comment]
Okay, I will :)
Why do you think that it shouldn't be a metric?
When you initialize the metric you calculate the statistics of the training set.
Then you perform updates according to all validation batches (for example), and then in the end compute the distance between the two.
This approach is also DDP "ready".

**Comment by ohayonguy** [comment]
https://github.com/PyTorchLightning/metrics/issues/169

Moved to torchmetrics.
Closing.

**Comment by awaelchli** [comment]
> Why do you think that it shouldn't be a metric?

No, not saying it shouldn't. It would of course be very useful as a metric. I encourage you to push for it.
